radicle-cli/src/commands/patch/common.rs

```diff
@@ -39,68 +37,26 @@ pub fn confirm<D: fmt::Display>(prompt: D, options: &Options) -> bool {
     !options.confirm || term::confirm(prompt)
 }
 
-/// List of merge targets.
-#[derive(Debug, Default)]
-pub struct MergeTargets {
-    /// Merge targets that have already merged the patch.
-    pub merged: Vec<Remote>,
-    /// Merge targets that haven't merged the patch.
-    pub not_merged: Vec<(Remote, git::Oid)>,
-}
-
-/// Find potential merge targets for the given head.
-pub fn find_merge_targets(
-    head: &Oid,
-    branch: &git::RefStr,
-    storage: &Repository,
-) -> anyhow::Result<MergeTargets> {
-    let mut targets = MergeTargets::default();
-
-    for remote in storage.remotes()? {
-        let (_, remote) = remote?;
-        let Some(target_oid) = remote.refs.head(branch) else {
-            continue;
-        };
-
-        if is_merged(storage.raw(), target_oid.into(), *head)? {
-            targets.merged.push(remote);
-        } else {
-            targets.not_merged.push((remote, target_oid));
-        }
-    }
-    Ok(targets)
-}
-
 /// Determine the merge target for this patch. This can ben any tracked remote's "default"
 /// branch, as well as your own (eg. `rad/master`).
 pub fn get_merge_target(
-    project: &Project,
     storage: &Repository,
     head_branch: &git::raw::Branch,
-) -> anyhow::Result<(storage::Remote, git::Oid)> {
-    let head_oid = head_branch
-        .get()
-        .target()
-        .ok_or(anyhow!("invalid HEAD ref; aborting"))?;
-    let mut spinner = term::spinner("Analyzing remotes...");
-    let targets = find_merge_targets(&head_oid, project.default_branch().as_refstr(), storage)?;
+) -> anyhow::Result<(String, git::Oid)> {
+    let spinner = term::spinner("Analyzing remotes...");
+    let (qualified_ref, target_oid) = storage.canonical_head()?;
+    let head_oid = branch_oid(head_branch)?;
+    let merge_base = storage.raw().merge_base(*head_oid, *target_oid)?;
+
+    if head_oid == merge_base.into() {
+        anyhow::bail!("changes have already been merged");
+    }
```

Comment:
This feels like it could belong in its own section for patch validation

Reply:
Yeah this makes sense here. I'd change the error though to something like `patch commits are already part of the canonical branch; nothing to do`